corehq/messaging/smsbackends/ivory_coast_mtn/models.py

```diff
@@ -0,0 +1,101 @@
+from corehq.apps.sms.models import SMS
+from corehq.apps.sms.util import strip_plus
+from corehq.util.timezones.conversions import ServerTime
+from lxml import etree
+
+
+class IvoryCoastMTNBackend(SQLSMSBackend):
+
+    class Meta:
+        app_label = 'sms'
+        proxy = True
+
+    @classmethod
+    def get_api_id(cls):
+        return 'IVORY_COAST_MTN'
+
+    @classmethod
+    def get_generic_name(cls):
+        return "Ivory Coast MTN"
+
+    @classmethod
+    def get_available_extra_fields(cls):
+        return [
+            'customer_id',
+            'username',
+            'password',
+            'sender_id',
+        ]
+
+    @classmethod
+    def get_form_class(cls):
+        return IvoryCoastMTNBackendForm
+
+    @classmethod
+    def get_ivory_coast_timestamp(cls):
+        return ServerTime(datetime.utcnow()).user_time(pytz.timezone('Africa/Abidjan')).done()
+
+    def get_params(self, msg_obj):
+        config = self.config
+
+        return {
+            'customerID': config.customer_id,
+            'userName': config.username,
+            'userPassword': config.password,
+            'originator': config.sender_id,
+            # This is for a custom project and they only send messages in French
+            'messageType': 'Latin',
+            # This must be set to the date and time to send the message; it's a required field
+            'defDate': self.get_ivory_coast_timestamp().strftime('%Y%m%d%H%M%S'),
+            'blink': 'false',
+            'flash': 'false',
+            'private': 'false',
+            'smsText': msg_obj.text.encode('utf-8'),
+            'recipientPhone': msg_obj.phone_number,
+        }
+
+    @classmethod
+    def phone_number_is_valid(cls, phone_number):
+        phone_number = strip_plus(phone_number)
+        # Phone number must be an Ivory Coast phone number
+        # Also avoid processing numbers that are obviously too short
+        return phone_number.startswith('225') and len(phone_number) > 3
+
+    def send(self, msg_obj, *args, **kwargs):
+        if not self.phone_number_is_valid(msg_obj.phone_number):
+            msg_obj.set_system_error(SMS.ERROR_INVALID_DESTINATION_NUMBER)
+            return
+
+        response = requests.get(
+            'http://smspro.mtn.ci/smspro/soap/messenger.asmx/HTTP_SendSms',
+            params=self.get_params(msg_obj),
+        )
+        self.handle_response(msg_obj, response.status_code, response.text)
+
+    def get_result_and_transaction_id(self, response_text):
+        root_tag = etree.fromstring(response_text.encode('utf-8'))
+        result_tag = root_tag.find('{http://pmmsoapmessenger.com/}Result')
+        transaction_id_tag = root_tag.find('{http://pmmsoapmessenger.com/}TransactionID')
+
+        return {
+            'result': result_tag.text if result_tag is not None else None,
+            'transaction_id': transaction_id_tag.text if transaction_id_tag is not None else None,
+        }
```

Comment:
If simple enough to add, it'd be nice to have a test for this response parsing

Reply:
yup, added https://github.com/dimagi/commcare-hq/pull/18546/commits/ea943f5680f4ddbb66d1b12a7027b304c2dd83cb and https://github.com/dimagi/commcare-hq/pull/18546/commits/b15099ffaaf88e6c3760ef3bcf931938f9e23f6e